### PR TITLE
KF unit test with in-situ simulation

### PR DESCRIPTION
**Pull request by beomki-yeo**
This PR makes the KF unit test run the simulation before reconstruction instead of utilizing pre-generated simulation data.
The motivation is obvious - it has been very painful to update the traccc-data whenever there is a change in the detray simulation.

### CODE REVIEW

**Comment by beomki-yeo** [comment]
Anyone has seen this warning? I included detray simulation in traccc KF test and this error appears. I don't know why it doesn't appear in `simulate_telescope.cpp` example :thinking: 

```
 /__w/traccc/traccc/build/_deps/dfelibs-src/dfe/dfe_io_dsv.hpp:323:6: error: 'particle' may be used uninitialized [-Werror=maybe-uninitialized]
  323 |   os << x;
      |   ~~~^~~~
```

**Comment by beomki-yeo** [comment]
It seems fixed after acts-project/detray#469

File: tests/sycl/test_kalman_filter.sycl [hunk 1]
```diff
@@ -38,9 +39,10 @@
 #include <gtest/gtest.h>
 
 // System include(s).
-#include <climits>
+// System include(s).
```
Comment (guilhermeAlmeida1) [inline tests/sycl/test_kalman_filter.sycl:42]: Remove this duplicate line

**Review by guilhermeAlmeida1 — COMMENTED** [review]
Trivial comments from my part

File: tests/common/tests/kalman_fitting_test.hpp [hunk 2]
```diff
@@ -12,17 +12,18 @@
...
+#include "detray/masks/masks.hpp"
+#include "detray/propagator/navigator.hpp"
+#include "detray/propagator/rk_stepper.hpp"
+#include "detray/simulation/event_generator/track_generators.hpp"
 
 // GTest include(s).
 #include <gtest/gtest.h>
 
 // System include(s).
+#include <limits>
```
[the remark this answers is not in the document]
Comment (beomki-yeo) [reply tests/common/tests/kalman_fitting_test.hpp:26]: Yeah I was using numeric_limits and removed it eventually.
I removed the header now

**Review by guilhermeAlmeida1 — APPROVED** [review]
LGTM

**Pull request merged**